### PR TITLE
Magicite

### PR DESCRIPTION
**Pull request by cocosolos**
Various fixes to 4-1 Magicite. Missing CS, misplaced CS, various NPC fixes.

### CODE REVIEW

File: scripts/zones/RuLude_Gardens/npcs/Goggehn.lua [hunk 1]
```diff
@@ -4,70 +4,60 @@
...
 -----------------------------------
-local ID = require("scripts/zones/RuLude_Gardens/IDs")
-require("scripts/globals/settings")
 require("scripts/globals/keyitems")
 require("scripts/globals/missions")
 -----------------------------------
 
 function onTrade(player,npc,trade)
-end
+end;
```
Comment (Cryptexx) [inline scripts/zones/RuLude_Gardens/npcs/Goggehn.lua]: No semicolons here please

File: scripts/zones/Altar_Room/Zone.lua [hunk 2]
```diff
@@ -12,6 +12,9 @@ end;
 
 function onZoneIn(player,prevZone)
     local cs = -1;
+    if (player:getVar("FickblixCS") == 1) then
```
Comment (Cryptexx) [inline scripts/zones/Altar_Room/Zone.lua]: In the majority of the code, you removed parenthesis, but added them here. No need for inner/outer parenthesis in these conditionals.

File: scripts/zones/RuLude_Gardens/npcs/_6r5.lua [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+require("scripts/globals/missions");
+-----------------------------------
+
+function onTrade(player,npc,trade)
+end;
+
+function onTrigger(player,npc)
+    pNation = player:getNation();
+    currentMission = player:getCurrentMission(pNation);
+    MissionStatus = player:getVar("MissionStatus");
```
Comment (Cryptexx) [inline scripts/zones/RuLude_Gardens/npcs/_6r5.lua]: These variables should be localized, e.g. local MissionStatus = ...

File: scripts/zones/RuLude_Gardens/npcs/_6r5.lua [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+function onEventUpdate(player,csid,option)
+end;
+
+function onEventFinish(player,csid,option)
+    if csid == 39 then
+        finishMissionTimeline(player,1,csid,option);
+    elseif csid == 130 and option == 1 then
+        player:setVar("MissionStatus",1)
+        if player:hasKeyItem(dsp.ki.ARCHDUCAL_AUDIENCE_PERMIT) == false then
```
Comment (Cryptexx) [inline scripts/zones/RuLude_Gardens/npcs/_6r5.lua]: Can just say: if not player:hasKeyItem(dsp.ki.ARCHDUCAL_AUDIENCE_PERMIT) then

File: scripts/zones/RuLude_Gardens/npcs/_6r2.lua [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+function onTrade(player,npc,trade)
+end
+
+function onTrigger(player,npc)
+    local pNation = player:getNation()
+
+    if pNation == dsp.nation.BASTOK then
+        local currentMission = player:getCurrentMission(pNation)
+        local MissionStatus = player:getVar("MissionStatus")
```
Comment (kaincenteno) [inline scripts/zones/RuLude_Gardens/npcs/_6r2.lua]: is MissionStatus needed? only see it being used once here 🐱

Comment (TeoTwawki) [reply scripts/zones/RuLude_Gardens/npcs/_6r2.lua]: while that is true, ima let it stay as is for width reasons

**Comment by TeoTwawki** [comment]
@cocosolos tested all the way through as all 3 nation?

**Comment by cocosolos** [comment]
@TeoTwawki Tested and works with all 3 nations

**Pull request merged**